docs/bugs.json

```diff
@@ -1,4 +1,14 @@
 [
+    {
+        "uid": "SOL-2021-3",
+        "name": "SignedImmutables",
+        "summary": "Immutable variables of signed integer type shorter than 256 bits can have non-cleaned higher order bits if they are accessed via inline assembly.",
+        "description": "When immutable variables of signed integer type shorter than 256 bits are read, their higher order bits will be zero. The correct operation would be to sign-extend the value, i.e. set the higher order bits to one if the sign bit is one. This sign-extension is performed by Solidity just prior to whern it matters, i.e. when a value is stored in memory, when it is compared or when a division is performed. Because of that, to our knowledge, the only way to access the value in its unclean state is by reading it through inline assembly.",
```

Comment:
```suggestion
        "description": "When immutable variables of signed integer type shorter than 256 bits are read, their higher order bits were incorrectly set to zero. The correct operation would be to sign-extend the value, i.e. set the higher order bits to one if the sign bit is one. This sign-extension is performed by Solidity just prior to when it matters, i.e. when a value is stored in memory, when it is compared or when a division is performed. Because of that, to our knowledge, the only way to access the value in its unclean state is by reading it through inline assembly.",
```